File: UIAutomation/Behave/pages/BasketPage.py
import random
from selenium.webdriver.common.by import By

from pages.CategoryPage import CategoryPage
from pages.HomePage import HomePage
from utils.Driver import Driver
from utils.Helper import Helper


class Elements:
    sepet_durum = (By.XPATH, "//span[contains(text(),'Sepetim (1 Ürün)')]")


class BasketPage:
    def __init__(self):
        self.driver = Driver.get()

    def sepete_gidildi_mi(self):
        if Helper.isDisplayed(Elements.sepet_durum):
            assert True
            print("Sepete başarıyla gidildi.")
        else:
            assert False, "Sepete gidilemedi."
            print("Sepete gidilemedi.")

File: UIAutomation/Cucumber/src/test/java/Case/pages/BasketPage.java
package Case.pages;

import Case.utils.Driver;
import Case.utils.Helper;
import Case.utils.Log4j;
import org.junit.Assert;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

import java.util.List;
import java.util.Random;

public class BasketPage extends Helper {
    private final Random ran;

    public BasketPage() {
        PageFactory.initElements(Driver.get(), this);
        ran = new Random();
    }

    @FindBy(xpath = "//span[contains(text(),'Sepetim (1 Ürün)')]")
    public WebElement sepetDurum;

    public void sepeteGidildiMi() {
        if (sepetDurum.isDisplayed()) {
            Assert.assertTrue(true);
            Log4j.info("Sepete başarıyla gidildi.");
        } else {
            Assert.fail("Sepete gidilemedi.");
            Log4j.error("Sepete gidilemedi.");
        }
    }
}
